### DISCUSSION TITLE
Download youtube clips

### DISCUSSION THREAD

**Discussion by eshref** [Tyrrrz/YoutubeDownloader#340] [Q&A] [answered]
Hi, first of all let me thank you for such an app with great performance and usability and helpful UI.

Currently I am using it to download videos from youtube then extract some portion of this video using free [shotcut](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiguYy_l67_AhU6hP0HHednBK8QFnoECA4QAQ&url=https%3A%2F%2Fshotcut.org%2F&usg=AOvVaw2xpjc0m9Y-157GByUGFK1x) software.

Then youtube introduced clips functionality with which one can create clips.
![image](https://github.com/Tyrrrz/YoutubeDownloader/assets/34239435/dc577e8b-a717-406c-bb09-410adc3f98d3)

Can we download this clips using YoutubeDownloader? Currently it is very time consuming downloading whole video (for example 3 hour) and cut special part.

Currently when I provide clip link app shows below behaviour:
![image](https://github.com/Tyrrrz/YoutubeDownloader/assets/34239435/2fc99275-1de3-46a5-bc61-d1ac0b855fbe)

Thanks in advance.

**Comment by Tyrrrz** [comment]
Hmm, I'll need to check if they share the same infrastructure as normal videos. Can you send me a link to a random clip?

**Reply by eshref** [reply]
For example: https://www.youtube.com/clip/UgkxA6JXuZQPfcOTR5uY1aob1WJNQQHrgLSO

**Reply by Tyrrrz** [reply, marked as answer]
It looks like the actual "clipping" is performed client-side – YouTube serves the full video page and then the browser seeks to the specified position in time. Essentially, clips are more or less the same as timed links (`&t=60`) and not a different type of videos.

It is possible to do the same on our end, but we'd need to perform some potentially complicated calculations to convert a timestamp into a byte offset, while accounting for different encoding formats. The easy way is to do the trimming after the fact with ffmpeg, but that would require downloading the entire video first.
